data/src/main/java/com/antosik/benchproject/data/common/api/MoviesModule.kt

```diff
@@ -9,32 +9,32 @@ import com.antosik.benchproject.domain.popular.movies.repo.PopularMoviesReposito
 import dagger.Module
 import dagger.Provides
 import dagger.hilt.InstallIn
-import dagger.hilt.components.SingletonComponent
-import javax.inject.Singleton
+import dagger.hilt.android.components.ViewModelComponent
 import retrofit2.Retrofit
 import retrofit2.converter.gson.GsonConverterFactory
 import retrofit2.create
 
 @Module
-@InstallIn(SingletonComponent::class)
-internal object MoviesModule {
+@InstallIn(ViewModelComponent::class)
+object MoviesModule {
 
-    @Singleton
     @Provides
     fun provideRetrofit(): Retrofit = Retrofit.Builder()
         .addConverterFactory(GsonConverterFactory.create())
         .baseUrl(Constants.BASE_URL)
         .build()
 
-    @Singleton
-    @Provides
-    fun provideMoviesApi(retrofit: Retrofit): MoviesApi = retrofit.create()
+    @Module
+    @InstallIn(ViewModelComponent::class)
+    internal object InternalMoviesModule {
```

Comment:
![image](https://user-images.githubusercontent.com/12697488/180409931-5feaa277-03ff-438f-b2e0-518cd4289399.png)

Reply:
I changed it on `Providers` since in this project I have 3 functions and one of them is for `moviesApi`

Reply:
Not perfect but better than before :)

Reply:
Ok :D